### PR TITLE
【Auto-Parallel | Comm】fix communication hang issue on the GPU-H

### PR DESCRIPTION
**Pull request by liym27**
### PR Category

Auto Parallel

### PR Types

Bug fixes

### Description

fix communication hang issue on the GPU-H

PCard-86802

### CODE REVIEW

File: test/auto_parallel/hybrid_strategy/pir_reshard_nd_mesh_cross_mesh.py [hunk 1]
```diff
@@ -104,10 +104,19 @@ def run_pp_to_rr_case(self):
         new_ops_name = [op.name() for op in dist_program.global_block().ops]
 
         rank_id = dist.get_rank()
+        print(f"[liyamei check] new_ops_name:\n{new_ops_name}")
```
Comment (zhiqiu) [inline test/auto_parallel/hybrid_strategy/pir_reshard_nd_mesh_cross_mesh.py]: remove debug code

File: test/auto_parallel/static_reshard_api_cross_mesh.py [hunk 2]
```diff
@@ -160,8 +161,10 @@ def test_reshard_mesh(self):
                 'pd_op.relu',
                 'pd_op.matmul',
                 'pd_op.add',
-                'pd_op.send_v2',
-                'pd_op.recv_v2',
+                'pd_op.share_data_',
+                'pd_op.broadcast',
+                'pd_op.full',
+                'pd_op.broadcast',
```
Comment (zhiqiu) [inline test/auto_parallel/static_reshard_api_cross_mesh.py:166]: does share_data_ and full affect the performance?

File: python/paddle/distributed/auto_parallel/static/reshard_funcs/same_status_reshard_func.py [hunk 3]
```diff
@@ -127,32 +120,37 @@ def reshard(self, src_dist_attr, dst_dist_attr, src_value, dst_type):
...
+            broadcast_op.dist_attr = (
+                paddle.base.libpaddle.pir.create_op_dist_attribute(
+                    bcast_mesh, [src_dist_attr], [dst_dist_attr], chunk_id
+                )
+            )
         if is_send:
             # fake var will be removed in remove_other_rank_op_pass.
             fake_var = paddle._C_ops.reshard_v2(src_value, dst_dist_attr)
             fake_var.set_type(dst_type)
             return fake_var
```
Comment (zhiqiu) [inline python/paddle/distributed/auto_parallel/static/reshard_funcs/same_status_reshard_func.py:163]: why still need fake_var?

File: test/auto_parallel/hybrid_strategy/pir_reshard_nd_mesh_cross_mesh.py (hunk 1 above)
Comment (liym27) [reply test/auto_parallel/hybrid_strategy/pir_reshard_nd_mesh_cross_mesh.py]: Thx. done

File: test/auto_parallel/static_reshard_api_cross_mesh.py (hunk 2 above)
Comment (liym27) [reply test/auto_parallel/static_reshard_api_cross_mesh.py:166]: Performance verification in progress

File: python/paddle/distributed/auto_parallel/static/reshard_funcs/same_status_reshard_func.py (hunk 3 above)
Comment (liym27) [reply python/paddle/distributed/auto_parallel/static/reshard_funcs/same_status_reshard_func.py:163]: > why still need fake_var?

If the current rank is not src rank or dst rank, fakevar is needed.

**Pull request closed without merging**